changelogs/unreleased/gh-8921-memory-leak-in-xrow_decode_error.md

```diff
@@ -0,0 +1,3 @@
+## bugfix/core
+
+* Fixed the memory leak on `error_unpack_unsafe()` failure (gh-8921).
```

Comment:
`error_unpack_unsafe` is an internal function unknown to the users. Better rephrase to something like: Fixed the memory leak on unpacking an invalid MsgPack error extension.

Reply:
Done.